app/controllers/account_activations_controller.rb

```diff
@@ -1,17 +1,31 @@
+# account activation controller
 class AccountActivationsController < ApplicationController
-
   def edit
     user = User.find_by(email: params[:email])
-    if user && !user.activated? && user.authenticated?(:activation, params[:id])
-      user.activate
-      user.update_attribute(:activated,    true)
-      user.update_attribute(:activated_at, Time.zone.now)
-      log_in user
-      flash[:success] = "Account activated!"
-      redirect_to user
+    if user && !user.activated?
+      if user.authenticated?(:activation, params[:id])
+        account_activation
```

Comment:
this should be called `activate_user` - a method's name should represent an action. Like - this method will `activate_user`.